CorsixTH/Lua/ui.lua

```diff
@@ -505,8 +505,17 @@ function UI:onKeyDown(rawchar, modifiers, is_repeat)
       self:onMouseDown(mapped_button, self.cursor_x, self.cursor_y)
       return true
     end
-    key = self.key_remaps[key] or key
+    if string.sub(key,1,6) ~= "keypad" or modifiers["numlockactive"] == nil then
+      key = self.key_remaps[key] or key
+    end
+  end
+
+  -- If numlock is active and the keypad is used remove the Keypad prefix
+  if string.sub(rawchar,1,6) == "Keypad" and modifiers["numlockactive"] then
+    rawchar = string.sub(rawchar,8)
```

Comment:
What happens if you put a condition here for ```numlockactive``` on and off, setting to numbers like you do when on, or arrows like you do in the file right now when off, then remove the condition you added above.

I think it would be more straight forward.